### PR TITLE
[Bugfix] ahrefs

### PR DESCRIPTION
**Pull request by MymmiJ**
> _NOTE TO REVIEWERS_ - please **do not review PRs in the `DRAFT` state**, as the PR may change substantially before it is ready to review. Thanks.

## Description

Stops pages breaking if any elements have a style applied.
Fixes links if they have punctuation.

### CODE REVIEW

**Review by snikidev — APPROVED** [review]

File: data/transform/pre-parser/textile-js-workarounds/fix-links.ts [hunk 1]
```diff
@@ -8,5 +8,10 @@ export const fixDuplicateQuoteLinks: StringTransformation = (content) => content
 export const fixHtmlElementsInLinks: StringTransformation = (content) =>
   content.replace(/"([^"<]+)":([^<\s]+)</gm, '<a href="$2">$1</a><');
 
+// Punctuation immediately after links is interpreted correctly by textile-js; but was not by Nanoc.
+// We need to remove it to retain parity.
+export const fixPunctuationInLinks: StringTransformation = (content) =>
```
Comment (aralovelace) [inline data/transform/pre-parser/textile-js-workarounds/fix-links.ts:13]: In my IDE it says - TS2451: Cannot redeclare block-scoped variable 'fixPunctuationInLinks'. ?

File: src/components/blocks/external-references/A.tsx [hunk 2]
```diff
@@ -22,7 +23,7 @@ const A = ({ data, attribs }: HtmlComponentProps<'a'>): ReactElement => {
     const imageElement = data.find((item) => item.name === 'img');
     // @ts-ignore - says src doesn't exist, but it does
```
Comment (aralovelace) [inline src/components/blocks/external-references/A.tsx:24]: hmm if you do?

```
const imageAttr = imageElement?.attribs as unknown as HTMLImageElement;
if (imageAttr?.src === rawHref && imageElement?.data) {
```

**Review by aralovelace — COMMENTED** [review]
LGTM just 2 minor comments

File: data/transform/pre-parser/textile-js-workarounds/fix-links.ts (hunk 1 above)
Comment (MymmiJ) [reply data/transform/pre-parser/textile-js-workarounds/fix-links.ts:13]: Sounds like an error in your IDE 🤷

**Pull request merged**